{"category": "ham", "to_address": "\"Talbot Katz\" <topkatz@msn.com>, r-help@stat.math.ethz.ch", "from_address": "\"Alberto Monteiro\" <albmont@centroin.com.br>", "subject": "Re: [R] Testing for existence inside a function", "body": "\nTalbot Katz wrote:\n> \n> I'm having trouble testing for existence of an object inside a function.\n> \nNo, you are having trouble testing for existence of an object\n_before_ the function is called :-)\n\n> Suppose I have a function:\n> \n> f<-function(x){\n> ...\n> }\n> \n> and I call it with argument y:\n> \n> f(y)\n> \n> I'd like to check inside the function whether argument y exists.\n>\nThis can't be done, because the error happens before f is called.\n\nTry this:\n\nf <- function(x) x + 1\nf(y.does.not.exist)\ny.does.not.exist\n\nThe error message is (almost) the same, and it happens when\nparsing the line. There's no way to change f to change this.\n\nAlberto Monteiro\n\n______________________________________________\nR-help@stat.math.ethz.ch mailing list\nhttps://stat.ethz.ch/mailman/listinfo/r-help\nPLEASE do read the posting guide http://www.R-project.org/posting-guide.html\nand provide commented, minimal, self-contained, reproducible code.\n\n"}